File: oop/lista01_java/KeyboardInput.java
package Lista01;

import java.io.InputStream;
import java.util.Scanner;

public class KeyboardInput {

    private final Scanner sc;

    public KeyboardInput(InputStream inputSource) {
         this.sc = new Scanner(inputSource);
    }

    public Integer readInteger() {
        while (!sc.hasNextInt()) {
            System.out.println("The input must be an integer.");
            sc.next();
        }
        return sc.nextInt();
    }

    public Integer readPositiveInteger() {
        Integer input;
        while (!sc.hasNextInt()) {
            System.out.println("The input must be an integer.");
            sc.next();
        }

        if ((input = sc.nextInt()) <= 0) throw new IllegalArgumentException("Number must be positive.");
        return input;
    }

    public Double readDouble() {
        while (!sc.hasNextDouble()) {
            System.out.println("The input must be a double.");
            sc.next();
        }
        return sc.nextDouble();
    }

    public String readString() {
        while (!sc.hasNextLine()) {
            System.out.println("The input must be a string.");
            sc.next();
        }
        return sc.nextLine();
    }
}

File: objetos/lista01_py/KeyboardInput.py
class KeyboardInput:
    def readPositiveInt(self, message = ""):
        value = input(message)
        isValidInput = False

        while not isValidInput:
            try:
                value = int(value)
                if value <= 0: raise Exception()
                isValidInput = True
            except:
                value = input("Your input must be a positive integer. Try again: ")
        return value

    def readFloat(self, message = ""):
        value = input(message)
        isValidInput = False

        while not isValidInput:
            try:
                value = float(value)
                isValidInput = True
            except:
                value = input("Your input must be a float number. Try again: ")
        return value